src/components/views/rooms/DecryptionFailureBar.tsx

```diff
@@ -74,11 +82,12 @@ export const DecryptionFailureBar: React.FC<IProps> = ({ failures, room }) => {
         }
     }, [needsVerification, hasOtherVerifiedDevices, failures, room, requestedSessions, context]);
 
+    // Recheck which devices are verified and whether we have key backups
     const updateDeviceInfo = useCallback(async () => {
-        context.crypto.deviceList.invalidateUserDeviceList(context.getUserId());
+        context.crypto.deviceList.invalidateUserDeviceList(context.getUserId()!);
         await context.crypto.deviceList.refreshOutdatedDeviceLists();
```

Comment:
please don't do this - it shouldn't be necessary, and is making my work in integrating the rust crypto-sdk harder by using the internal `crypto` object.

If the device list isn't being correctly updated, let's fix that, not just arbitrarily invalidate it.

Reply:
This isn't a matter of the device list not being correctly updated; it's that we need new data right now, and we're invalidating it because we know it's out-of-date.

Reply:
how do we know that?

Reply:
References to crypto internals have been removed, we're now downloading device keys through the proper API methods to update information after we do things that affect our device keys/verification status.